app/src/main/java/org/mozilla/focus/tiles/HomeTile.kt

```diff
@@ -0,0 +1,68 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.tiles
+
+import org.json.JSONObject
+import java.util.UUID
+
+private const val KEY_URL = "url"
+private const val KEY_TITLE = "title"
+
+private const val KEY_IMG = "img"
+private const val KEY_ID = "id"
+
+sealed class HomeTile(val url: String, val title: String) {
+    protected fun appendToJSONObject(obj: JSONObject) {
+        obj.put(KEY_URL, url)
+        obj.put(KEY_TITLE, title)
+    }
+}
+
+class DefaultHomeTile(
+        url: String,
+        title: String,
+        val imagePath: String?,
+        /** Unique id used to identify specific home tiles, e.g. for deletion, etc. **/
+        val id: String
+) : HomeTile(url, title) {
+
+    fun toJSONObject(): JSONObject {
+        val obj = JSONObject(mapOf(
+                KEY_IMG to imagePath,
+                KEY_ID to id
+        ))
+        super.appendToJSONObject(obj)
+        return obj
+    }
+
+    companion object {
+        fun fromJSONObject(jsonObject: JSONObject): DefaultHomeTile {
+            return DefaultHomeTile(jsonObject.getString(KEY_URL),
+                    jsonObject.getString(KEY_TITLE),
+                    jsonObject.optString(KEY_IMG),
+                    jsonObject.getString(KEY_ID))
+        }
+    }
+}
+
+class CustomHomeTile(
+        url: String,
+        title: String,
+        val id: UUID
```

Comment:
nit: Comment that UUID is used for screenshots, unless you add that later in your commit.